wsmaster/codenvy-hosted-api-organization/src/main/java/com/codenvy/organization/api/resource/OrganizationResourcesLimitManager.java

```diff
@@ -0,0 +1,286 @@
+import com.codenvy.resource.api.usage.ResourceUsageManager;
+import com.codenvy.resource.api.usage.ResourcesLocks;
+import com.codenvy.resource.model.Resource;
+import com.google.common.annotations.VisibleForTesting;
+
+import org.eclipse.che.api.core.ConflictException;
+import org.eclipse.che.api.core.NotFoundException;
+import org.eclipse.che.api.core.Page;
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.workspace.server.StripedLocks;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.emptyList;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Facade for organization resources redistribution related operations.
+ *
+ * @author Sergii Leschenko
+ */
+@Singleton
+public class OrganizationResourcesLimitManager {
+    private final OrganizationResourcesLimitDao organizationResourcesLimitDao;
+    private final ResourcesLocks                resourcesLocks;
+    private final ResourceUsageManager          usageManager;
+    private final ResourceAggregator            resourceAggregator;
+    private final OrganizationManager           organizationManager;
+
+    @Inject
+    public OrganizationResourcesLimitManager(OrganizationResourcesLimitDao organizationResourcesLimitDao,
+                                             ResourcesLocks resourcesLocks,
+                                             ResourceUsageManager usageManager,
+                                             ResourceAggregator resourceAggregator,
+                                             OrganizationManager organizationManager) {
+        this.organizationResourcesLimitDao = organizationResourcesLimitDao;
+        this.resourcesLocks = resourcesLocks;
+        this.usageManager = usageManager;
+        this.resourceAggregator = resourceAggregator;
+        this.organizationManager = organizationManager;
+    }
+
+    /**
+     * Set resources limit for suborganization on parent's resources usage.
+     *
+     * @param resourcesLimit
+     *         resources limit to store
+     * @throws NullPointerException
+     *         when either {@code resourcesLimit} is null
+     * @throws NotFoundException
+     *         when organization with {@code resourcesLimit.getId()} was not found
+     * @throws ConflictException
+     *         when parent organization doesn't have enough resources to increase limit
+     * @throws ConflictException
+     *         when limit can't be changed because suborganization is using existed resources
+     *         or when they are redistributed to next organizations level
+     * @throws ServerException
+     *         when any other error occurs
+     */
+    public void setResourcesLimit(OrganizationResourcesLimit resourcesLimit) throws NotFoundException,
```

Comment:
The name of the method looks inconsistent with get, i mean it's should be either
```java
setResourcesLimit 
getResourcesLimit
```

or(i like this one)
```
setLimit
getLimit
```

or event

```
set
get
```
but not 
```
setResourcesLimit
get
```